src/fetch.ts

```diff
@@ -165,21 +162,14 @@ export function createFetch(globalOptions: CreateFetchOptions = {}): $Fetch {
         }
       }
     }
-
-    let abortTimeout: NodeJS.Timeout | undefined;
-
-    // TODO: Can we merge signals?
-    if (!context.options.signal && context.options.timeout) {
-      const controller = new AbortController();
-      abortTimeout = setTimeout(() => {
-        const error = new Error(
-          "[TimeoutError]: The operation was aborted due to timeout"
-        );
-        error.name = "TimeoutError";
-        (error as any).code = 23; // DOMException.TIMEOUT_ERR
-        controller.abort(error);
-      }, context.options.timeout);
-      context.options.signal = controller.signal;
+    if (typeof context.options.timeout === "number") {
+      context.options.signal = AbortSignal.any(
+        [
+          AbortSignal.timeout(context.options.timeout),
```

Comment:
Moving from `setTimeout` to `AbortSignal.timeout` is a new requirement (**even** if it is supported in baseline browser versions) we need to use `setTimeout` when there is no custom `context.options.signal`

Reply:
Ohh. So if someone passes a signal, we know signals are supported? I can adjust this :)